sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/CliSuite.scala

```diff
@@ -159,6 +160,17 @@ class CliSuite extends SparkFunSuite with BeforeAndAfterAll with Logging {
     }
   }
 
+  test("Pick spark.sql.warehouse.dir first for Spark Cli if set") {
+    val sparkWareHouseDir = Utils.createTempDir()
+    new File(warehousePath, "metastore_db").delete()
+    runCliWithin(
+      1.minute,
+      Seq("--conf", s"""${StaticSQLConf.WAREHOUSE_PATH.key}=$sparkWareHouseDir"""))(
+      """desc database default;""" -> sparkWareHouseDir.getAbsolutePath
+    )
+    sparkWareHouseDir.delete()
```

Comment:
how do we verify if the warehouse path is picked?

Reply:
the hive.xxx.xxx.dir is from the global variable `warehousePath`, if the `sparkWareHouseDir` not picked, the db location of default will be the hive one

Reply:
I also checked manually
#### now
```sql
kentyao@hulk  ~/Downloads/spark/spark-3.1.0-SNAPSHOT-bin-v20200317  bin/spark-sql -S --conf spark.sql.warehouse.dir=/Users/kentyao/Downloads/spark/spark-3.1.0-SNAPSHOT-bin-v20200317/data/spark-warehouse
spark-sql> desc database default;
Database Name	default
Comment	Default Hive database
Location	file:/Users/kentyao/Downloads/spark/spark-3.1.0-SNAPSHOT-bin-v20200317/data/spark-warehouse
Owner	public
```
#### 3.x and 2.4
```sql

kentyao@hulk  ~/Downloads/spark/spark-2.4.5-bin-hadoop2.7  bin/spark-sql -S -c spark.sql.warehouse.dir=/Users/kentyao/Downloads/spark/spark-3.1.0-SNAPSHOT-bin-v20200317/data/spark-warehouse
spark-sql> desc formatted default;
Error in query: Table or view 'default' not found in database 'default';
spark-sql> desc database default;
Database Name	default
Description	Default Hive database
Location	file:/user/hive/warehouse
```

Reply:
How is it verified?

Reply:
In this test, we will run a spark-sql job to check the output
```
[info] - Pick spark.sql.warehouse.dir first for Spark Cli if set *** FAILED *** (1 minute)
[info]   =======================
[info]   CliSuite failure output
[info]   =======================
[info]   Spark SQL CLI command line: ../../bin/spark-sql --master local --driver-java-options -Dderby.system.durability=test --conf spark.ui.enabled=false --hiveconf javax.jdo.option.ConnectionURL=jdbc:derby:;databaseName=/Users/kentyao/spark/target/tmp/spark-3570629a-23a7-4e2a-b21a-9bce68025228;create=true --hiveconf hive.metastore.warehouse.dir=/Users/kentyao/spark/target/tmp/spark-dbe4dd6a-1129-43f5-83bc-62dd2072b410 --hiveconf hive.exec.scratchdir=/Users/kentyao/spark/target/tmp/spark-9422968c-94d0-4251-b591-30cbcc061845 --hiveconf conf1=conftest --hiveconf conf2=1 --conf spark.sql.warehouse.dir=/Users/kentyao/spark/target/tmp/spark-1219faf2-918f-4ac4-8d28-fa4dd2d24c7e
[info]   Exception: java.util.concurrent.TimeoutException: Futures timed out after [1 minute]
[info]   Executed query 0 "desc database default;",
[info]   But failed to capture expected output "/Users/kentyao/spark/target/tmp/spark-1219faf2-918f-4ac4-8d28-fa4dd2d24c7e" within 1 minute.
```